app/grandchallenge/archives/forms.py

```diff
@@ -225,3 +227,44 @@ def save(self, *args, **kwargs):
             {"interface_pk": self.cleaned_data["interface"].pk}
         )
         return super().save(*args, **kwargs)
+
+
+class ArchiveItemCreateForm(MultipleCIVForm):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields["title"] = CharField(
+            required=False,
+            initial=self.instance and self.instance.title or "",
+            max_length=ArchiveItem._meta.get_field("title").max_length,
+        )
+
+    class Meta:
+        non_civ_fields = ("title",)
+
+    def clean_title(self):
+        title = self.cleaned_data.get("title")
+        if title and self._title_query(title).exists():
+            raise ValidationError(
+                "An archive item already exists with this title"
+            )
+        return title
+
+    def _title_query(self, title):
+        return ArchiveItem.objects.filter(
+            title=title,
+            archive=self.base_obj,
+        )
```

Comment:
I think this could be refactored as well.

Reply:
Done!